Initial Kinesalite docker image

mikemoate opened the pull request:
A docker image to provide Kinesalite for development and testing, in lieu of AWS Kinesis Streams.

mikemoate committed:
Initial Dockerfile for Kinesalite

mikemoate committed:
Initial Docs

randomjunk commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+FROM node:8
```
any reason not `node:8-alpine` here? (just in the interests of smaller image size)

mikemoate replied:
Now done, though was more challenging than expected...

randomjunk commented on Dockerfile:
```diff
@@ -0,0 +1,13 @@
+USER kinesalite
+
+EXPOSE 4567
+ENTRYPOINT ["kinesalite", "--path", "/var/kinesalite"]
```
Do we need to specify path here?
If the primary purpose of the library is correctness testing I wouldn't have thought there would be much data passing through, and the keep-in-RAM mode would be faster.

Path can always be added at the run stage if people need it.

mikemoate replied:
No and fair point, removed.

randomjunk commented on README.md:
```diff
@@ -0,0 +1,51 @@
+# Kinesalite
+
+[![Docker Automated buil](https://img.shields.io/docker/build/countingup/kinesalite.svg)](https://hub.docker.com/r/countingup/kinesalite/builds/)
```
missing buil`d` ?

mikemoate committed:
Use alpine based node image

This was more complex than expected, since kinesalite depends on leveldown which is a binary package.

Need to force this to be built from source (alpine has musl not glibc) and need to work around npm permissions issues with installing global packages as root.

mikemoate committed:
Don't use filesystem for data by default

In memory will be more performant for testing, and can always specify `--path` if this is desired.

mikemoate committed:
Update README for changes

randomjunk reviewed (approved):
LGTM
